packages/server/src/course/course.controller.ts

```diff
@@ -59,10 +63,20 @@ export class CourseController {
       .getRawMany();
     course.heatmap = await this.heatmapService.getCachedHeatmapFor(id);
 
-    course.queues = await async.filter(
-      course.queues,
-      async (q) => await q.checkIsOpen(),
-    );
+    const userCourseModel = await UserCourseModel.findOne({
+      where: {
+        user,
+        courseId: id,
+      },
+    });
+
+    if (userCourseModel.role !== Role.PROFESSOR) {
+      course.queues = await async.filter(
+        course.queues,
+        async (q) => await q.checkIsOpen(),
+      );
+    }
```

Comment:
Would it make sense to have the filter allow `isProfessorQueue` queues if the user is a professor instead of ignoring the filter all together? 

That way we can avoid having a large number of queues showing up for a professor when we switch back to in-person office hours.

Reply:
that makes a lot more sense...